Sprint 14

Kolesnikov-Eugene opened the pull request.

Kolesnikov-Eugene committed:
HomeViewController
AddTrackerViewController
Models

Kolesnikov-Eugene committed:
all UIelements of HomeView is ready
AddTrackerViewController modified (label, buttons)
TabBarController created with HomeView and StatisticsView(empty)

Kolesnikov-Eugene committed:
Tracker, TrackerCategory and TrackerRecord models added.
HomeViewCollectionViewCell UI ready.
Header added to collectionView.

Kolesnikov-Eugene committed:
NewHabitViewController: full UI except dynamic errorLabel presentation(TODO)
ScrollView working in habitView
AddScheduleViewController(TODO tableView)

Kolesnikov-Eugene committed:
ScheduleViewController added to the project

Kolesnikov-Eugene committed:
AddScheduleDelegate
AddScheduleViewController saves UISwitch state from AddScheduleViewCell
SingleSelection in collectionView in each section in NewHabitViewController

Kolesnikov-Eugene committed:
extension Date to correctly filter trackers
trackerDoneButton full functionality
Filter by datePicker

Kolesnikov-Eugene committed:
SearchBar is working
Added irreguralEvent creation option
Refactoring
full functionality of the project!

TODO: refactror

Kolesnikov-Eugene committed:
Search bar working properly, when backspacing field
refactoring

amardanov commented on Tracker/Model/Tracker.swift:
```diff
@@ -0,0 +1,148 @@
+    }
+}
+
+enum Schedule: Int, CaseIterable {
```
Это перечисление лучше вынести в отдельный файл для большего удобства и уменьшения файла `Tracker.swift`

amardanov commented on Tracker/TabBar/TabBarController.swift:
```diff
@@ -0,0 +1,50 @@
+        self.viewControllers = [homeViewNavigationController, statisticViewController]
+    }
+}
+
```
Лишние 3 строчки остались

amardanov commented on Tracker/View/AddScheduleView/AddScheduleViewCell.swift:
```diff
@@ -0,0 +1,103 @@
+        return view
+    }()
+    var callBackSwitchState: ((Bool) -> (Void))?
+//    var switchState: Bool?
```
Закомментированный код лучше не оставлять в коде приложения. Так код будет чище и ревьюеру будет проще проверять за счет уменьшения количества строк

amardanov commented on Tracker/View/AddScheduleView/AddScheduleViewController.swift:
```diff
@@ -0,0 +1,139 @@
+                return
+            }
+            
+            if isOn {
```
Можно записать в одну строку с помощью турнирного оператора
```
isOn ? selectedDays.append(day) :  selectedDays.removeAll(where: { $0 == day })
```

amardanov commented on Tracker/View/AddScheduleView/AddScheduleViewController.swift:
```diff
@@ -0,0 +1,139 @@
+        return cell
+    }
+    
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
```
Не забывай про форматирование кода в проекте. Отступы можно поправить с помощью сочетаний клавиш `cmd+a`, `control+i`. Советую пройтись и это сделать в каждом файле

Kolesnikov-Eugene committed:
Completed trackers are now stored in completedTrackers variable.
Refactored methods of fulfilling trackers.
Deleted fields (date, counter, completedAt) from Tracker model.
Changed all the fonts among the project according to design in Figma.
Added emtySearch result image and label.
Tracker can be marked as completed for passed days.
Refactoring among the whole project

Kolesnikov-Eugene committed:
Category and typeTracker fields removed from Tracker model.
Changed method of adding a new tracker.
